sdk/java/src/main/java/com/gojek/feast/RequestUtil.java

```diff
@@ -17,47 +17,88 @@
 
 import feast.serving.ServingAPIProto.FeatureReference;
-import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 @SuppressWarnings("WeakerAccess")
 public class RequestUtil {
 
+  /**
+   * Create feature references protos from given string feature reference.
+   *
+   * @param featureRefStrings to create Feature Reference protos from
+   * @param project specifies to the project set in parsed Feature Reference protos otherwise ""
+   * @return List of parsed {@link feast.serving.ServingAPIProto.FeatureReference} protos
+   */
   public static List<FeatureReference> createFeatureRefs(
-      List<String> featureRefStrings, String defaultProject) {
+      List<String> featureRefStrings, String project) {
     if (featureRefStrings == null) {
       throw new IllegalArgumentException("featureRefs cannot be null");
     }
 
-    List<FeatureReference> featureRefs = new ArrayList<>();
+    List<FeatureReference.Builder> featureRefs =
+        featureRefStrings.stream()
+            .map(refStr -> parseFeatureRef(refStr, false))
+            .collect(Collectors.toList());
+    // apply project override if specified
+    if (!project.isEmpty()) {
+      featureRefs =
+          featureRefs.stream().map(ref -> ref.setProject(project)).collect(Collectors.toList());
+    }
 
-    for (String featureRefString : featureRefStrings) {
-      String project;
-      String name;
-      String[] projectSplit = featureRefString.split("/");
+    return featureRefs.stream().map(ref -> ref.build()).collect(Collectors.toList());
+  }
 
-      if (projectSplit.length == 1) {
-        project = defaultProject;
-        name = projectSplit[0];
-      } else if (projectSplit.length == 2) {
-        project = projectSplit[0];
-        name = projectSplit[1];
+  /**
+   * Parse a feature reference proto builder from the given featureRefString
+   *
+   * @param featureRefString string feature reference to parse from.
+   * @param ignoreProject whether to ignore project when parsing.
+   * @return a parsed {@link FeatureReference.Builder}
+   */
+  public static FeatureReference.Builder parseFeatureRef(
+      String featureRefString, boolean ignoreProject) {
+    featureRefString = featureRefString.trim();
+    if (featureRefString.isEmpty()) {
+      throw new IllegalArgumentException("Cannot parse a empty feature reference");
+    }
+    FeatureReference.Builder featureRef = FeatureReference.newBuilder();
+
+    // parse project if specified
+    if (featureRefString.contains("/")) {
+      if (ignoreProject) {
+        String[] projectSplit = featureRefString.split("/");
+        featureRefString = projectSplit[1];
       } else {
         throw new IllegalArgumentException(
-            String.format(
-                "Feature id '%s' has invalid format. Expected format: <project>/<feature-name>.",
-                featureRefString));
+            String.format("Unsupported feature reference: %s", featureRefString));
```

Comment:
how does `ignoreProject` relate to the incoming request. It seems like `project1/feature_set:feature` would raise an exception every time if `ignoreProject` is true.

Reply:
Clarified.